docs/how-to-migrate-from-smartlock.md

```diff
@@ -0,0 +1,791 @@
+          AuthenticationMethods.EMAIL,
+          AuthenticationMethods.PHONE,
+          AuthenticationMethods.GOOGLE)
+          .setTokenProviders(tokenProviders)
+          .build());
+
+  startActivityForResult(hintIntent, RC_HINT);
+}
+```
+
+The key differences to note are:
+
+- The `CredentialClient` is easier to create than a `GoogleApiClient` instance:
+  a `Context` reference is all that is required, with no asynchronous
+  initialization.
+
+- Account types from the Smart Lock for Passwords API are now “authentication
+  methods”, with email auth, phone auth and federated authentication methods
+  all treated in a consistent manner.
+
+- A plain `Intent` is used instead of a `PendingIntent` - we have found this to
+  be less confusing for developers, and the additional functionality provided by
+  PendingIntent proved to be unnecessary.
+
+One may also notice the new `setTokenProviders` method - in an effort to support
+acquiring ID tokens from additional providers, clients are now encouraged to
+explicitly declare the ID token providers they support (e.g. Google,
+Microsoft), along with any provider-specific parameters that may be required.
+Currently, OpenYOLO directly supports specifying your application’s OpenID
+Connect client ID and an ID token nonce value.
+
+### Receiving a hint response
+
+Hint responses are delivered to the application via onActivityResult in both
+the Smart Lock and OpenYOLO APIs. Both indicate the success or failure of the
+request via a result code and Intent extras, though the data is structurally
+different.
+
+With Smart Lock for Passwords:
+
+```java
+@Override protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT && resultCode == RESULT_OK) {
+    Credential credential =
+        data.getParcelableExtra(Credential.EXTRA_KEY);
+    if (credential == null) {
+      authenticateManually();
+    } else {
+      processSmartLockCredentialAsHint(credential);
+    }
+  }
+}
+```
+
+With OpenYOLO:
+
+```java
+@Override
+protected void onActivityResult(
+    int requestCode, int resultCode, Intent data) {
+
+  if (requestCode == RC_HINT) {
+    HintRetrieveResult result =
+        mCredentialClient.getHintRetrieveResult(data);
+
+    if (result.getResultCode() !=
+        HintRetrieveResult.CODE_HINT_SELECTED) {
+      authenticateManually();
+      return;
+    }
+
+    Hint hint = result.getHint();
+    processOpenYoloHint(hint);
+  }
+}
+```
+
+With OpenYOLO, the `resultCode` value will always be equal to the value returned
+by `result.getResultCode()`, so either can be used. The standard result codes
+are  all defined as constants on the `HintRetrieveResult` class. The result
+codes in OpenYOLO are more detailed than those returned by Smart Lock for
+Passwords, providing more information particularly in the cases where a hint is
+not returned:
+
+|----------------------------------|-------------------------------------------|
```

Comment:
It seems that the table is broken? I was taking a look on preview mode.
https://github.com/iainmcgin/OpenYOLO-Android/blob/fc4740915cbe9460e015ddc0b77d2b2fd6c4f29f/docs/how-to-migrate-from-smartlock.md

![screen shot 2017-10-09 at 09 45 34](https://user-images.githubusercontent.com/2204411/31328673-9ee49b2e-acd6-11e7-9a7d-237d15865051.png)